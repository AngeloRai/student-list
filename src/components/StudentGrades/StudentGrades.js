
const StudentGrades = ({index, grade}) => {
  return (
    <div className=" text-sm ">
      Test {index + 1}: &nbsp;&nbsp;&nbsp;&nbsp;{grade}%
    </div>
  );
};

export default StudentGrades;
