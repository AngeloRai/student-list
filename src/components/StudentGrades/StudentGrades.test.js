import React from "react";
import { shallow, mount } from "enzyme";
import StudentGrades from "./StudentGrades";
import toJson from "enzyme-to-json";

describe("tests grades component", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<StudentGrades />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("accepts props", () => {
    const mockIndex = "1";
    const mockGrade = "75";
    const wrapper = mount(
      <StudentGrades index={mockIndex} grade={mockGrade} />
    );
    expect(wrapper.props().index).toBe(mockIndex);
    expect(wrapper.props().grade).toBe(mockGrade);
  });
});
