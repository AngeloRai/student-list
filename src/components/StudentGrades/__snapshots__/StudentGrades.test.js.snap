// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests grades component renders without crashing 1`] = `
<div
  className=" text-sm "
>
  Test 
  NaN
  :     
  %
</div>
`;
